src/nix/install.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+# Move to the same directory as this script
+set -e
+FEATURE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "${FEATURE_DIR}"
+
+# Option defaults
+VERSION="${VERSION:-"latest"}"
+MULTIUSER="${MULTIUSER:-"true"}"
+PACKAGES="${PACKAGES:-""}"
+DERIVATIONPATH="${DERIVATIONPATH:-""}"
+FLAKEURI="${FLAKEURI:-""}"
+USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"
+
+# Nix keys for securly verifying installer download signature per https://nixos.org/download.html#nix-verify-installation
+NIX_GPG_KEYS="B541D55301270E0BCF15CA5D8170B4726D7198DE"
+GPG_KEY_SERVERS="keyserver hkp://keyserver.ubuntu.com:80
+keyserver hkps://keys.openpgp.org
+keyserver hkp://keyserver.pgp.com"
+
+if [ -e "/nix" ]; then
+    echo "(!) Nix is already installed! Aborting."
```

Comment:
I know this is a pattern we are using for our other Features, which probably needs to be changed.

As this is a brand new feature, can we check if the already installed version of nix is >= to the requested version ? If so, then exit else install the requested version.

Reply:
I don't think that is a good idea in this case. You can't have multiple versions of nix installed like you could something like node. 

Installing from scratch wouldn't work without impacting was was already present. You likely would need to delete the nix folder in its entirety along with all the packages to be able to safely switch, which could have the unintended consequence of deleting something the user expects. I suspect we'd have to delete all the users created as well for the install script to succeed.

So the only other option is to try to perform an upgrade, but that is not as straight forward as it sounds and it varies based on whether the original install was multi-user or not... so we'd need to start tracking it. See https://nixos.org/manual/nix/stable/installation/upgrading.html

So, lots of complexity for questionable pay-off here.

I could skip the install but still try to install the specified packages or just fail outright if either of those are preferrable.

Reply:
Oh - also the Nix install is like 600mb... and doing an upgrade or reinstall would land in a new layer which could ad another 600mb to the image. Generally this is something you should do once, but I could see allowing the packages/flake part to still execute.

Reply:
😮  wasn't aware of these complexities. Thanks for the detailed explanation.

> I could skip the install but still try to install the specified packages 

If we could do that, then it would be great & should sync with our existing pattern 👍  (eg. python: we install tools even if some python is already installed - details [here](https://github.com/devcontainers/features/blob/main/src/python/install.sh#L404))

Reply:
@samruddhikhandale Ok, done!